PurchasesCoreSwift/Misc/DateExtensions.swift

```diff
@@ -5,10 +5,14 @@
 
 import Foundation
 
+enum DateExtensionsError: Error {
+    case notValidDateFromComponents
```

Comment:
nit: I think `invalidDateComponents` could be a better name

Reply:
good call

Reply:
I'm sorry I didn't catch this before, I just realized it might be useful to capture the date components that trigger this and spit those out to aid folks in debugging. Thoughts?

Reply:
love it, adding

Reply:
done! lmk what you think

Reply:
Love it. 🎈🐐